kernel/src/table_changes/physical_to_logical.rs

```diff
@@ -79,62 +44,361 @@ pub(crate) fn scan_file_physical_schema(
     }
 }
 
+/// Creates a CDF-specific transform expression by combining partition values with CDF metadata.
+/// This is the CDF-specific wrapper around the generic get_transform_expr function.
+pub(crate) fn get_cdf_transform_expr(
+    transform_spec: &[FieldTransformSpec],
+    partition_values: &HashMap<String, String>,
+    logical_schema: &StructType,
+    scan_file: &CdfScanFile,
+) -> DeltaResult<ExpressionRef> {
+    // Build partition values separately
+    let partition_field_values =
```

Comment:
I think we can just `parse_partition_values_to_scalars` here and get rid of `parse_partition_values_to_expressions` all together. and then just `into` the value below. Saves an iteration of the map and reduces overall code